src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -365,7 +365,16 @@ internal NamespaceOrTypeOrAliasSymbolWithAnnotations BindNamespaceOrTypeOrAliasS
                 case SyntaxKind.PredefinedType:
                     {
                         var type = BindPredefinedTypeSymbol((PredefinedTypeSyntax)syntax, diagnostics);
-                        return TypeSymbolWithAnnotations.CreateNonNull(NonNullTypes, type);
+                        if (type.SpecialType == SpecialType.System_Boolean)
+                        {
+                            // Breaking a cycle when binding the `NonNullTypes` attribute constructor
+                            // PROTOTYPE(NullableReferenceTypes): test this with some extra/unrelated constructors on NonNullTypesAttribute
+                            return TypeSymbolWithAnnotations.CreateNonNull(nonNullTypes: true, type);
```

Comment:
Seems a little odd that we have a BinderFlag specifically to avoid cycles, but we also need to do another thing to break cycles. This feels like it could be fragile, and it also feels like if we're trying to special case the `NonNullTypes` attribute constructor this is too wide of a `if` condition.

Reply:
This special case will probably have to be removed or fixed differently in a follow-up PR. The test with an extra constructor on `NonNullTypesAttribute` crashes.